### PR TITLE
Misc: bug fixes 

### PR DESCRIPTION
**Pull request by sanket1729**
Misc bug fixes

### CODE REVIEW

**Review by dr-orlovsky — DISMISSED** [review]
utACK 88d3bbcb2a4e729be9fde8d8571616d0c2fbc6cf

**Review by dr-orlovsky — APPROVED** [review]
utACK 32e918687b1f9a02a4562d10032450e004c9fcb7

File: src/policy/semantic.rs [hunk 1]
```diff
@@ -413,18 +413,16 @@ impl<Pk: MiniscriptKey> Policy<Pk> {
...
-                            if is_or {
-                                ret_subs.extend(or_subs);
+                        Policy::Threshold(k, subs) => {
+                            if is_and && is_or {
+                                // means m = n = 1, thresh(1,X) type thing.
+                                ret_subs.push(Policy::Threshold(k, subs));
+                            } else if k == 1 && is_or {
+                                ret_subs.extend(subs)
+                            } else if k == subs.len() && is_and {
+                                ret_subs.extend(subs)
```
[the remark this answers is not in the document]
Comment (sanket1729) [reply src/policy/semantic.rs]: Done

**Review by apoelstra — APPROVED** [review]
ACK 549f6a10384644b951738e81304609b80553a9ae

The PR description should be changed to reflect that this isnt' MultiA support.

**Pull request merged**